src/main.rs

```diff
@@ -19,49 +18,53 @@ mod process_manager;
 mod router;
 mod user_config;
 
-use rocket::Rocket;
-
 use crate::{
     addon_manager::{AddonManager, LoadAddons},
     db::Db,
 };
 use actix::{System, SystemService};
-use std::thread;
+use rocket::{Build, Rocket};
 
-fn rocket() -> Rocket {
-    rocket::ignite()
+fn rocket() -> Rocket<Build> {
+    rocket::build()
         .manage(Db::new())
         .mount("/", router::routes())
 }
 
-fn main() -> () {
-    let system = System::new("default");
+#[rocket::main]
+async fn main() {
+    tokio::spawn(async {
+        let system = System::new("default");
 
-    actix::spawn(async move {
-        addon_socket::start().await.expect("Starting addon socket");
-    });
+        actix::spawn(async {
+            addon_socket::start().await.expect("Starting addon socket");
+        });
 
-    actix::spawn(async move {
-        AddonManager::from_registry()
-            .send(LoadAddons)
-            .await
-            .expect("Sending LoadAddons message ");
-    });
+        actix::spawn(async {
+            AddonManager::from_registry()
+                .send(LoadAddons)
+                .await
+                .expect("Sending LoadAddons message ");
+        });
 
-    thread::spawn(|| {
-        rocket().launch();
+        system.run().expect("Running system");
```

Comment:
This is a blocking call.
I think we need to pull that out of the `spawn` call.
I'll do it after the PR is merged.